src/main/kotlin/inc/evil/d17map/web/filters/GlobalExceptionHandlingFilter.kt

```diff
@@ -0,0 +1,62 @@
+package inc.evil.d17map.web.filters
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import inc.evil.d17map.exceptions.ErrorCodes
+import inc.evil.d17map.exceptions.ErrorResponse
+import inc.evil.d17map.exceptions.JWTFilterException
+import io.github.oshai.kotlinlogging.KotlinLogging
+import jakarta.servlet.FilterChain
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.core.Ordered
+import org.springframework.core.annotation.Order
+import org.springframework.stereotype.Component
+import org.springframework.web.filter.OncePerRequestFilter
+
+
+private val kotlinLogger = KotlinLogging.logger {}
+
+@Order(Ordered.HIGHEST_PRECEDENCE)
+@Component
+class GlobalExceptionHandlingFilter(private val objectMapper: ObjectMapper) : OncePerRequestFilter() {
+
+    override fun doFilterInternal(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        filterChain: FilterChain
+    ) {
+        try {
+            filterChain.doFilter(request, response)
+        } catch (ex: Exception) {
+            handleException(response, ex)
+        }
+    }
+
+    private fun handleException(
+        response: HttpServletResponse,
+        ex: Exception,
+    ) {
+        kotlinLogger.error(ex) { "Exception caught in GlobalExceptionHandlingFilter: ${ex.message}" }
+
+        val errorResponse = when (ex) {
+            is JWTFilterException -> {
+                response.status = HttpServletResponse.SC_UNAUTHORIZED
+                ErrorResponse(
+                    errorCode = ErrorCodes.UNAUTHORIZED_ERROR,
+                    message = "${ex.message}"
+                )
+            }
+            else -> {
+                response.status = HttpServletResponse.SC_INTERNAL_SERVER_ERROR
+                ErrorResponse(
+                    errorCode = ErrorCodes.UNKNOWN_ERROR,
+                    message = "Internal Server Error: ${ex.message}"
+                )
+            }
+        }
+
+        response.contentType = "application/json"
+        val jsonResponse = objectMapper.writeValueAsString(errorResponse)
+        response.writer.write(jsonResponse)
```

Comment:
`````
try {
    val jsonResponse = objectMapper.writeValueAsString(errorResponse)
    response.writer.write(jsonResponse)
} catch (serializationEx: Exception) {
    kotlinLogger.error(serializationEx) { "Error serializing ErrorResponse" }
    response.writer.write("{\"error\":\"Internal Server Error\"}")
}

Reply:
i może warto byłoby zrobić jakieś zabezpieczenie przed ponownym zapisaniem, coś w stylu` if (!response.isCommitted)`

Reply:
Co do 1 tooo się zastanawiam, bo może się zdarzyć, że z jakiegoś powodu failuje object mapper i bez tego catche'a zostanie zwrócony Internal Server Error, ale nie będzie miał naszej struktury błędu, więc może akurat to byłoby dobrze dodać.

Reply:
Co do 2 to nie rozumiem 😅

Reply:
Co do 2 zastanawiam się, czy faktycznie taka sytuacja może mieć tutaj miejsce. Chodzi o to, że czasami np w wyniku jakichś błędów, response został już zatwierdzony i wysłany do klienta i mimo to probójemy na przykład wysłać jeszcze raz/modyfikować - to miałoby zabezpieczyć przed taką sytuacją - ale na chwilę obecną w sumie nie wiem czy taka sytuacja tutaj mogłaby mieć miejsce. Może się zbyt czepiam xd zostaw to jednak bez tego

Reply:
Tzn. Jeśli jesteś w tym filtrze, to odpowiedź nie mogła zostać wysłana, bo wysłanie odpowiedzi znaczy, że wyszłaś już z filtrów. Nie wiem co by się musiało stać, żebyś wracała z rekurencji i jednocześnie już z niej wróciła xD Filtr Schrödingera